# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/02 09:45:50-08:00 akpm@osdl.org 
#   [PATCH] sparc64 pmd_offset() fix
#   
#   mm/memory.c: In function `zeromap_pud_range':
#   mm/memory.c:1053: warning: suggest parentheses around + or - inside shift
#   mm/memory.c: In function `remap_pud_range':
#   mm/memory.c:1170: warning: suggest parentheses around + or - inside shift
#   
#   Parenthesize this macro arg.
#   
#   Cc: Nick Piggin <nickpiggin@yahoo.com.au>
#   Signed-off-by: Andrew Morton <akpm@osdl.org>
#   Signed-off-by: Linus Torvalds <torvalds@osdl.org>
# 
# include/asm-sparc64/pgtable.h
#   2005/01/02 01:36:40-08:00 akpm@osdl.org +1 -1
#   sparc64 pmd_offset() fix
# 
diff -Nru a/include/asm-sparc64/pgtable.h b/include/asm-sparc64/pgtable.h
--- a/include/asm-sparc64/pgtable.h	2005-01-04 18:20:52 -08:00
+++ b/include/asm-sparc64/pgtable.h	2005-01-04 18:20:52 -08:00
@@ -315,7 +315,7 @@
 /* Find an entry in the second-level page table.. */
 #define pmd_offset(dir, address)	\
 	((pmd_t *) pgd_page(*(dir)) + \
-	 ((address >> PMD_SHIFT) & (REAL_PTRS_PER_PMD-1)))
+	 (((address) >> PMD_SHIFT) & (REAL_PTRS_PER_PMD-1)))
 
 /* Find an entry in the third-level page table.. */
 #define pte_index(dir, address)	\
